rsa.go

```diff
@@ -43,7 +44,17 @@ func (pk *RsaPublicKey) Raw() ([]byte, error) {
 }
 
 func (pk *RsaPublicKey) Encrypt(b []byte) ([]byte, error) {
-	return rsa.EncryptPKCS1v15(rand.Reader, pk.k, b)
+	partLen := pk.k.N.BitLen()/8 - 11
+	chunks := split([]byte(b), partLen)
+	buffer := bytes.NewBuffer([]byte(""))
+	for _, chunk := range chunks {
+		bytes, err := rsa.EncryptPKCS1v15(rand.Reader, pk.k, chunk)
```

Comment:
this is not gonna cut it, you make chunks and apply padding. That's not the way to encrypt large messages with RSA -- you typically create a symmetric encryption key, encrypt that with RSA, and use it to encrypt the message.

Reply:
The reason why i cut message into chunk is if we use `EncryptPKCS1v15` to encrypt message,The message must be no longer than the length of the public modulus minus 11 bytes.

You can see the note above the function:
https://github.com/golang/go/blob/bf8e6b70276ddfd3e070964542020fe782c1eec8/src/crypto/rsa/pkcs1v15.go#L29-L39

And it has caused bug in [go-ipfs #5536](https://github.com/ipfs/go-ipfs/issues/5536).